### PR TITLE
Update CFP to clarify a couple of details

### PR DESCRIPTION
**Pull request by drewbrew**
- Looking for a full bid rather than an hourly rate
- Include a working code base as a goal but not a hard requirement in the deliverables

Thanks to @Scotchester for the feedback!

### CODE REVIEW

**Review by PeterGrand — APPROVED** [review]

File: _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md
Comment (jefftriplett) [inline _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md]: We should probably list the schedule list, schedule details, presenter details, and updated open graph previews since these pages have different layouts.

Comment (drewbrew) [reply _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md]: Good call. Let me tweak this slightly

Comment (drewbrew) [reply _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md]: Added a separate bulleted list in the deliverables section

File: _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md
Comment (jefftriplett) [inline _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md:74]: I'm struggling with this wording still. We are asking for the full website design (side note: we should spell out this is an HTML + associated files), but then I think "ideally" makes the rest of this come off as optional. 

For a proposal, optional means we shouldn't expect to get that, and I guess my question is what we really want? cc @sodevious

**Review by jefftriplett — CHANGES_REQUESTED** [review]
I left a question to clarify what our deliverable is.

File: _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md
Comment (sodevious) [reply _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md:74]: I'm open to either-or, depending on the proposal. I'd like to hear what they can do

Comment (drewbrew) [reply _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md:74]: I'd like to maybe put it as a "strongly preferred" but leaving it open to still being an option if someone comes up with a sublime design that we can implement ourselves without too much work

File: _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md
Comment (Scotchester) [inline _posts/2024-01-12-call-for-proposals-for-djangocon-us-2024-website.md]:
```suggestion
	* Blog list
	* Blog details
* Open Graph previews for schedule details and presenters
```

**Review by Scotchester — COMMENTED** [review]
Thanks, y'all! I made one minor formatting suggestion, but I have one more broad suggestion:

Given that the location has now been announced as being in Durham again, I think it would be good to say more about how much you want to differentiate the 2024 site/branding from 2023.

I'd say the existence of this CFP implies that you don't want something as basic as swapping out the color palette or changing the font, but is there any aspect of the 2023 design that you WANT to maintain some continuity with, or would tolerate if it saved money? If the new branding is compatible with it, would you be opposed to maintaining the same basic page layouts, perhaps even using the code from 2023 as a basis?

Maybe I'm overthinking it and the simple answer is that you're open to all of these possibilities, and people should just propose the kind of project they want to engage in? :) But if any of the above questions make you realize that there are certain dealbreakers that you definitely have, it might be good to mention those.

Cheers!

**Comment by jefftriplett** [comment]
> I'd say the existence of this CFP implies that you don't want something as basic as swapping out the color palette or changing the font, but is there any aspect of the 2023 design that you WANT to maintain some continuity with, or would tolerate if it saved money? If the new branding is compatible with it, would you be opposed to maintaining the same basic page layouts, perhaps even using the code from 2023 as a basis?
> 
> Maybe I'm overthinking it and the simple answer is that you're open to all of these possibilities, and people should just propose the kind of project they want to engage in? :) But if any of the above questions make you realize that there are certain dealbreakers that you definitely have, it might be good to mention those.

@Scotchester thank you for the feedback. 

We have a few pain points that make working on the existing website harder than it needs to be. I'll list those in the order they annoy me:

### Design and layout changes require an experienced CSS developer to make them. 

**Upfront note:** None of these are anyone's fault. These are just the nature of these ecosystems. 

We have had a few times where we thought we could copy and paste a section from the top of a page to a lower in the page layout, but it would completely break. Our website uses best CSS design practices and requires a non-trivial amount of JS/SASS/CSS knowledge to make something like that work. 

Another pain point is that our Foundation version is just old enough, and our `package.json` requirements grow just stale enough that we can't just `npm install` and expect it to work. 

**Potential solution:** Move from Foundation to TailwindCSS, which is easier to "copy and paste" HTML and is better documented. TailwindCSS isn't a magic fix-all for us either, but they have tons of options from a CDN that builds style in the browser, static binary, and plenty of npm support, which gives us more flexibility. 

**What we need:** We, at minimum, need HTML and related design files for this. Ideally, 

### Local development is a pain

We optimized our local setup for Docker + Containers, which only some are comfortable running and using. This is a high bar/non-trivial learning curve if you have never worked with containers before. 

The website can be run without being in a container, but that requires installing Ruby and working with Gems, and we don't keep it updated. It's actually counter-intuitive to maintain because we pin to the GitHub Pages versions that GitHub uses, which doesn't give us a Gemfile. 

**Potential solution:** Embrace GitHub's Codespaces so that the website can be developed in-browser without local setup pains. 

**What we need:** Advice, and if someone has already tackled this, it would be a big bonus. I suspect we'll have to solve this ourselves. 

### Should we be using Jekyll? 

We picked GitHub Pages years ago because it was the least friction way of getting contributions from everyone. Anyone could edit their bio or make a copy contribution by clicking the edit button and submitting a pull request without knowing git. None of us love Jekyll, and there are significant pains we have learned to program around. 

GitHub Pages are still the default for GitHub, but we now have GitHub Actions which means we have options we didn't have ~5 years ago. 

**Potential solution:** We could switch to Eleventy, Astro, or something else, but see the previous points about changes that "require an experienced CSS developer to make them" apply to shifting from HTML templates to something that is JS writing out HTML. 

**What we need:** Advice and potential porting expertise. If someone has experience with other tools and understands our needs, we could lump this into the proposal or even be open to a follow-up proposal. 

### Big Picture (because this isn't long enough) 

In the long term, these websites outlive the conference and should be maintainable. Website locations/uris move, people go through life changes, and we find mistakes that warrant fixes. So, we need to be able to make changes and trust that the website will rebuild and not go offline.

**Comment by Scotchester** [comment]
@jefftriplett That's all very helpful information on the technical side, for sure. The part I wrote that you quoted, however, was actually more about the design. Basically, how different do you want the site to look from last year's?

I agree that moving away from Jekyll would be wise at this point. It feels like GitHub is letting their GitHub Pages Jekyll apparatus stagnate. It's pretty easy these days, as you suggest, to set up another static site generator and use GitHub Actions to rebuild on push. And using something else, anything that doesn't require Ruby, is going to go a long way to solving the local development pain point, as well.

As an old-timey CSS enthusiast, I have a hard time embracing the Tailwind utility class mindset, but I completely understand how it would be helpful in this scenario. That said, certain things like "We have had a few times where we thought we could copy and paste a section from the top of a page to a lower in the page layout, but it would completely break." could potentially be tackled by writing the CSS in a more scoped component-driven way. (I say this without having looked at the existing HTML/CSS, so take that with a grain of salt.)

---

Another thing that occurred to me is that, given this is for an event that is going to happen on a specific day, I would add a note about your desired delivery deadline, and how firm you want that to be.

**Comment by drewbrew** [comment]
> Basically, how different do you want the site to look from last year's?

For me at least, I'd like to leave that up to the designer. I don't want to say "the has to 100% be different" or "this should be almost exactly the same only with a new color scheme" or anything like that. We haven't really had much in the way of visual continuity except for the same typeface in the wordmark logo for a few years in a row and mostly reusing the same theme for 2021-22, and I kind of like having the fresh styling each year.

**Comment by sodevious** [comment]
> Basically, how different do you want the site to look from last year's?

Here you can see the difference in branding between years in San Diego:

https://2022.djangocon.us/
https://2019.djangocon.us/
https://2018.djangocon.us/


---

I would really hesitate to super define what the exact visual changes should be like. It's a creative process, and a proposal would initiate a conversation between all of us. 

That being said, updated branding is pretty important and the **main** goal of this proposal, as it affects all of our printed material, swag etc. We are open to hear proposals that are little code changes and mostly illustration, full branding and code base rebuilds, and everything in between.

I personally pushed for us to do a heavier code update this year, as I feel I will act as a resource in this process, and as a front-end developer by trade I have some opinions about how it can be done. But if we end up not doing a huge rebuild, that's fine. It's to spark a discussion between us and the person/persons proposing the project. Us mentioning specific things like Tailwind are not meant to be a 100% set in stone decision. As a fellow CSS enthusiast, I agree that component driven CSS is ideal... but we are in a situation where we have lots of maintainers of all experience levels, and limited time to make on-the-fly and ongoing updates.

**Comment by tim-schilling** [comment]
I'd vote to merge this changes because we're more likely to re-use the committed changes in the future rather than the changes in a closed PR.

**Pull request merged**
